### PR TITLE
[v1.7] Add NCCL_ASYNC_ERROR_HANDLING to docs (#46856)

### PR DESCRIPTION
**Pull request by osalpekar**
Summary:

Docs Changes for NCCL Async Error Handling. Adds info about the async error handling environment variable to the docs for the `init_process_group` function.
PR already landed to master: https://github.com/pytorch/pytorch/pull/46856

### CODE REVIEW

File: torch/distributed/distributed_c10d.py [hunk 1]
```diff
@@ -368,7 +368,20 @@ def init_process_group(backend,
...
-            is set to 1.
+            or ``NCCL_ASYNC_ERROR_HANDLING`` is set to 1. When
+            ``NCCL_BLOCKING_WAIT`` is set, this is the duration for which the
+            process will block and wait for collectives to complete before
+            throwing an exception. When ``NCCL_ASYNC_ERROR_HANDLING`` is set,
+            this is the duration after which collectives will be aborted
+            asynchronously and the process will crash. ``NCCL_BLOCKING_WAIT``
+            will provide errors to the user which can be caught and handled,
+            but due to its blocking nature, it has a performance overhead. On
+            the other hand, ``NCCL_ASYNC_ERROR_HANDLING`` has little
```
Comment (pritamdamania87) [inline torch/distributed/distributed_c10d.py:379]: There should be no perf overhead with `NCCL_ASYNC_ERROR_HANDLING` right? If so, should we update the comment to mention that. If not, let's just replace `little` with `very little` to emphasis not having a significant overhead.

File: torch/distributed/distributed_c10d.py [hunk 2]
```diff
@@ -368,7 +368,20 @@ def init_process_group(backend,
...
+            performance overhead, but crashes the process on errors. This is
+            done since CUDA execution is async and it is no longer safe to
+            continue executing user code since failed async NCCL operations
+            might result in subsequent CUDA operations to run on corrupted
+            data. Only one of these two environment variables should be set.
```
Comment (pritamdamania87) [inline torch/distributed/distributed_c10d.py:384]: Do we check this is the case and have a unit test that validates this? If not, would be good to address this in a separate PR.

**Review by pritamdamania87 — APPROVED** [review]

**Review by jspisak — APPROVED** [review]

File: torch/distributed/distributed_c10d.py (hunk 2 above)
Comment (osalpekar) [reply torch/distributed/distributed_c10d.py:384]: Sure, I can follow up later and add a test with both enabled

File: torch/distributed/distributed_c10d.py (hunk 1 above)
Comment (osalpekar) [reply torch/distributed/distributed_c10d.py:379]: It's not 0 perf overhead since there is a slight QPS regression measured, and it slowly closes the gap in each iteration, so I didn't want to be too absolute and say "no perf overhead".

**Review by seemethere — APPROVED** [review]
